### ISSUE TITLE
Generated targets cause problems upstream

### ISSUE THREAD

**Issue by adayton1** [LLNL/Umpire#802]
**Describe the bug**

```
CMake Error in src/main/CMakeLists.txt:
  Target "main" contains relative path in its
  INTERFACE_INCLUDE_DIRECTORIES:

    "include/umpire"
```

**To Reproduce**

Not sure exactly what triggers this, but I'm seeing similarly weird errors when building Axom with the latest Umpire.

**Additional context**

I'm pretty sure the error is somehow coming from the generator expressions here (we do have Fortran turned off):

```
target_include_directories(
  umpire_interface
  PUBLIC
  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src>
  $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>
  $<BUILD_INTERFACE:$<$<COMPILE_LANGUAGE:Fortran>:${CMAKE_Fortran_MODULE_DIRECTORY}>>
  $<INSTALL_INTERFACE:$<$<COMPILE_LANGUAGE:Fortran>:include/umpire>> # for Fortran module files
  $<INSTALL_INTERFACE:include>)
```

**Comment by adayton1** [comment]
I'm using CMake 3.21.1.

**Comment by adayton1** [comment]
This is probably a different issue, but it still comes from Umpire and may be related: https://github.com/LLNL/axom/issues/989

**Comment by adayton1** [comment]
```
INTERFACE_INCLUDE_DIRECTORIES "/usr/tce/packages/mvapich2/mvapich2-2.3.6-intel-oneapi.2022.2/include;/g/g17/dayton8/ale3d/update_raja_stack/lib/toss3_intel.oneapi.2022/camp/2022.10.1.a/compiler-intel-oneapi.2022.2-gcc-10.2.1/cpp-14/blt-0.5.2.a/cmake-3.21.1/cuda-none/include;${_IMPORT_PREFIX}/include;\$<\$<COMPILE_LANGUAGE:Fortran>:include/umpire>;${_IMPORT_PREFIX}/include"
```

This comes from the generated umpire-targets.cmake file. I would have thought that the generator expression would have been evaluated by this point, or there would have been a full path rather than a relative path.

**Comment by adayton1** [comment]
Here is the relevant portion, I believe:

`\$<\$<COMPILE_LANGUAGE:Fortran>:include/umpire>`

**Comment by adayton1** [comment]
We have disabled Fortran in Umpire, so this shouldn't even be in the generated umpire-targets.cmake file.

**Comment by adayton1** [comment]
Another data point. Using "$<$<COMPILE_LANGUAGE:Fortran>:$<INSTALL_INTERFACE:include/umpire>>" as the generator expression takes care of the relative path:

`\$<\$<COMPILE_LANGUAGE:Fortran>:${_IMPORT_PREFIX}/include/umpire>`

**Comment by adayton1** [comment]
Here's the fix I'm thinking about:

```
target_include_directories(
  umpire_interface
  PUBLIC
  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src>
  $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>
  $<INSTALL_INTERFACE:include>)

if (UMPIRE_ENABLE_FORTRAN)
  target_include_directories(
    umpire_interface
    PUBLIC
    $<$<COMPILE_LANGUAGE:Fortran>:$<BUILD_INTERFACE:${CMAKE_Fortran_MODULE_DIRECTORY}>>
    $<$<COMPILE_LANGUAGE:Fortran>:$<INSTALL_INTERFACE:include/umpire>>) # for Fortran module files
endif()
```

So if Fortran is disabled in Umpire, no generator expressions end up in the include directories. But if Fortran is enabled in Umpire, the generator expression is still there so the include directories are only added when building Fortran code.

**Comment by adayton1** [comment]
Fixed in MR #805
